File: src/main/java/um/haberes/core/model/view/LiquidacionPeriodo.java
package um.haberes.core.model.view;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;
import um.haberes.core.model.AuditableEntity;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Entity
@Table(name = "vw_liquidacion_periodo")
@Immutable
@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class LiquidacionPeriodo extends AuditableEntity {

    @Id
    private Long liquidacionId = null;

    private Long legajoId = null;

    private int anho = 0;

    private int mes = 0;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssXX", timezone = "UTC")
    private OffsetDateTime fechaLiquidacion = null;

    private Integer dependenciaId = null;

    private String salida = null;

    private BigDecimal totalRemunerativo = BigDecimal.ZERO;

    private BigDecimal totalNoRemunerativo = BigDecimal.ZERO;

    private BigDecimal totalDeduccion = BigDecimal.ZERO;

    private BigDecimal totalNeto = BigDecimal.ZERO;

    private Byte bloqueado = 0;

    private Long periodo = null;
}

File: src/main/java/um/haberes/core/kotlin/model/view/LiquidacionPeriodo.kt
package um.haberes.core.kotlin.model.view

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.*
import org.hibernate.annotations.Immutable
import um.haberes.core.kotlin.model.Auditable
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(name = "vw_liquidacion_periodo")
@Immutable
data class LiquidacionPeriodo(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var liquidacionId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fechaLiquidacion: OffsetDateTime? = null,

    var dependenciaId: Int? = null,
    var salida: String? = null,
    var totalRemunerativo: BigDecimal = BigDecimal.ZERO,
    var totalNoRemunerativo: BigDecimal = BigDecimal.ZERO,
    var totalDeduccion: BigDecimal = BigDecimal.ZERO,
    var totalNeto: BigDecimal = BigDecimal.ZERO,
    var bloqueado: Byte = 0,
    var periodo: Long? = null

) : Auditable()
